src/Shared/PackageManagers/PyPIProjectManager.cs

```diff
@@ -429,5 +436,34 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
 
             return mapping;
         }
+
+        public record PyPIArtifactUri : BaseArtifactUri
+        {
+            /// <summary>
+            /// Initializes a new instance of <see cref="PyPIArtifactUri"/>.
+            /// </summary>
+            /// <param name="type">The type of artifact for this <see cref="PyPIArtifactUri"/>.</param>
+            /// <param name="uri">The <see cref="Uri"/> this artifact can be found at.</param>
+            /// <param name="extension">The file extension for the file found at the <paramref name="uri"/>.</param>
+            public PyPIArtifactUri(PyPIArtifactType type, Uri uri, string? extension = null) : 
+                base(type, uri, extension ?? Path.GetExtension(uri.AbsolutePath))
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of <see cref="PyPIArtifactUri"/>.
+            /// </summary>
+            /// <param name="type">The type of artifact for this <see cref="PyPIArtifactUri"/>.</param>
+            /// <param name="uri">The string of the uri this artifact can be found at.</param>
+            /// <param name="extension">The file extension for the file found at the <paramref name="uri"/>.</param>
+            public PyPIArtifactUri(PyPIArtifactType type, string uri, string? extension = null) : this(type, new Uri(uri), extension) { }
+        }
+
+        public enum PyPIArtifactType
+        {
+            Unknown = 0,
+            Tarball,
+            Wheel,
```

Comment:
I don't see that you produce artifacts with this value.

Reply:
I had written it in preparation for eventually using it, but yeah I wasn't planning on using it in this PR. Older PyPI packages didn't include a `.whl` file so I wasn't sure if I wanted to construct them without confidence that it would exist.

Reply:
I can't find a way to construct the `.whl` file uris. I put a TODO down to figure it out.